common:
    lid_topic:  "/livox/repub"
    imu_topic:  "/livox/imu"
    time_sync_en: false         # ONLY turn on when external time synchronization is really not possible

preprocess:
    lidar_type: 1                # 1 for Livox serials LiDAR, 2 for Velodyne LiDAR, 3 for ouster LiDAR, 
    scan_line: 4
    blind: 2

mapping:
    acc_cov: 0.1
    gyr_cov: 0.1
    b_acc_cov: 0.0001
    b_gyr_cov: 0.0001
    fov_degree:    360.0
    det_range:     450.0
    extrinsic_est_en:  false      # true: enable the online estimation of IMU-LiDAR extrinsic
    extrinsic_T: [ -0.011, -0.0234, 0.044 ]
    extrinsic_R: [ 1, 0, 0,
                   0, 1, 0,
                   0, 0, 1]

irobot:
    # extrinsic_T: [ -0.16, -0.00, 0 ]
    # extrinsic_R: [ 0.7071, 0, -0.7071,
    #                0, 1, 0,
    #                0.7071, 0, 0.7071]
    # extrinsic_T: [ -0.18, 0, 0 ]
    # extrinsic_R: [ 1, 0, 0,
    #                0, 1, 0,
    #                0, 0, 1]
    extrinsic_T: [ -0.2, 0.5, 0.5 ]
    extrinsic_R: [0.9961947, -0.08715574, 0,
                  0.08715574, 0.9961947,  0,
                  0, 0, 1]
    # extrinsic_T: [0, 0, -0.32]
    # extrinsic_R: [ 1, 0, 0,
    #                0, -1, 0,
    #                0, 0, -1]

icp:
    use_icp: true
    save_result: true
    pub_result: true

    map_pcd_path: "/home/niuoruo/workspace/sentry/ws/sentry_navigation/src/faster-lio/PCD/scans.pcd"
    ori_pcd_path: "/home/niuoruo/workspace/sentry/ws/sentry_navigation/src/faster-lio/PCD/origin.pcd"
    res_pcd_path: "/home/niuoruo/workspace/sentry/ws/sentry_navigation/src/faster-lio/PCD/result.pcd"

    # ICP定位参数
    SCORE_THRESHOLD_MAX: 0.05     # 达到最大迭代次数后，代价仍高于此值，认为无法收敛
    ANGLE_UPPER_THRESHOLD_: 10    # 最大角度变换
    Point_Quantity_THRESHOLD: 500 # 点云数阈值
    Maximum_Iterations: 300        # 最大迭代次数
    VoxelGridRemoval_LeafSize: 0.1        # 体素滤波的边长
    ObstacleRemoval_Distance_Max: 1.0     # 如果雷达点云中点在地图点云最近点大于此值，就认为该点为障碍点

publish:
    path_en:  false
    scan_publish_en:  true       # false: close all the point cloud output
    dense_publish_en: true       # false: low down the points number in a global-frame point clouds scan.
    scan_bodyframe_pub_en: true  # true: output the point cloud scans in IMU-body-frame

pcd_save:
    pcd_save_en: false
    interval: -1                 # how many LiDAR frames saved in each pcd file; 
                                 # -1 : all frames will be saved in ONE pcd file, may lead to memory crash when having too much frames.

feature_extract_enable: false
point_filter_num: 3
max_iteration: 3
filter_size_surf: 0.5
filter_size_map: 0.5             # 暂时未用到，代码中为0， 即倾向于将降采样后的scan中的所有点加入map
cube_side_length: 1000

ivox_grid_resolution: 0.5        # default=0.2
ivox_nearby_type: 18             # 6, 18, 26
esti_plane_threshold: 0.1        # default=0.1